### DISCUSSION TITLE
square format for data matrix?

### DISCUSSION THREAD

**Discussion by FabioEasylogix** [micjahn/ZXing.Net#529] [Q&A] [answered]
Hello together, 

is it possible to use square data matrix format? It sometimes changes to rectangle format depending to the input string.

We use same sizeX/Y and data matrix format
ZXing.BarcodeFormat f = BarcodeFormat.DATA_MATRIX

if we generate an bitmap with the Write Method of the BarcodeWriter it has rectangle format, for strings in this format:

**AB12-CD3-E4**

and if we only use numbers like this 1234-567-89 it is working and the datamarix is square size.

Can anyone help me to find a solution to get always same square data matrix format?

**Comment by micjahn** [comment, marked as answer]
You should try the following code snippet:

```
var writer = new BarcodeWriter
{
    Format = BarcodeFormat.DATA_MATRIX,
    Options = new DatamatrixEncodingOptions
    {
        SymbolShape = Encoder.SymbolShapeHint.FORCE_SQUARE
    }
};
```

**Reply by FabioEasylogix** [reply]
Thank you after changing the options, it is working with square size.
